java/src/org/openqa/selenium/grid/node/config/CapabilitiesMutatorService.java

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.grid.node.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.openqa.selenium.Capabilities;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class CapabilitiesMutatorService {
+  Capabilities stereotype;
+  private final List<CapabilityMutator> customMutators = new CopyOnWriteArrayList<>();
```

Comment:
I think the `customMutators` is expected to be populated as soon as the object gets created. After that Its only going to be iterated upon. So do we still need a `CopyOnWriteArrayList`? Can we not just use `ArrayList` instead ?